src/node/lattice.rs

```diff
@@ -157,7 +159,8 @@ impl PartialOrd for RunnableEvent {
 /// ```
 pub struct ExecutionLattice {
     /// The `forest` is the directed acyclic graph that maintains the dependency graph of the
-    /// events.
+    /// events. The relation A -> B means that A *depends on* B.This dependency also indicates that
+    /// B precedes A (B < A) in the ordering. An event can be executed if it has no outbound edges.
     forest: Arc<Mutex<StableGraph<Option<OperatorEvent>, ()>>>,
```

Comment:
Can we actually skip the whole `Reversed` thing on each `add_events` by just changing this to
`forest: Arc<Mutex<Reversed<StableGraph<Option<OperatorEvent>, ()>>>>`?